public/js/pimcore/asset/image.js

```diff
@@ -427,30 +444,48 @@ pimcore.asset.image = Class.create(pimcore.asset.asset, {
 
         var markerDD = new Ext.dd.DD(marker);
 
-        this.marker = marker;
-    },
+        Ext.getCmp('remove_focal_point_' + this.id).setVisible(true);
+        Ext.getCmp('add_focal_point_' + this.id).setVisible(false);
 
-    getSaveData : function ($super, only) {
-        var parameters = $super(only);
+        this.marker = marker;
 
+    },
+    removeFocalPoint: function () {
         if(this["marker"]) {
+            this.marker.remove();
+            this["marker"] = false;
+            Ext.getCmp('remove_focal_point_' + this.id).setVisible(false);
+            Ext.getCmp('add_focal_point_' + this.id).setVisible(true);
+        }
+    },
+    getFocalPointCoordinates: function () {
+        let x = -100;
+        let y = -100;
 
-            var top = intval(this.marker.getStyle('top'));
-            var left = intval(this.marker.getStyle('left'));
+        if (this["marker"]) {
+            let top = intval(this.marker.getStyle('top'));
+            let left = intval(this.marker.getStyle('left'));
 
-            var boundingBox = this.marker.up().getSize();
+            let boundingBox = this.marker.up().getSize();
 
-            var x = round(left * 100 / boundingBox.width, 8);
-            var y = round(top  * 100 / boundingBox.height, 8);
+            x = round(left * 100 / boundingBox.width, 8);
+            y = round(top * 100 / boundingBox.height, 8);
+        }
+        this.focalPointCoordinates = {x: x, y: y};
```

Comment:
Set this only when recalculation happens as the property is initialized with same default values. then also get rid of x & y defaults.